src/Platform/Microsoft.Testing.Platform/TestHostControllers/TestHostControllersManager.cs

```diff
@@ -183,10 +183,10 @@ internal async Task<TestHostControllerConfiguration> BuildAsync(ServiceProvider
             // Check if we have already built the singleton for this composite factory
             if (!_alreadyBuiltServices.Contains(compositeServiceFactory))
             {
-                if (lifetimeHandlers.Any(x => x.TestHostProcessLifetimeHandler.Uid == extension.Uid))
+                (ITestHostProcessLifetimeHandler testHostProcessLifetimeHandler, int order)? existingHandler = lifetimeHandlers.Where(x => x.TestHostProcessLifetimeHandler.Uid == extension.Uid).FirstOrDefault();
```

Comment:
We also want to throw a useful localized message if there are more than one results.